src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3222,9 +3245,8 @@ void ApplicationManagerImpl::UnregisterApplication(
 
   commands_holder_->Clear(app_to_remove);
 
-  if (audio_pass_thru_active_) {
+  if (EndAudioPassThru(app_id)) {
```

Comment:
`app_id` is separate from `connection_key` correct. `connection_key` is used everywhere else.

Reply:
My understanding is that "app_id" and "connection_key" are eventually same.
- app_id is acquired through ApplicationImpl::app_id()
- The app_id_ value of ApplicationImpl is set through its constructor
- ApplicationImpl instance is created in ApplicationManagerImpl::RegisterApplication(). app_id is generated by session observer's `GetDataOnSessionKey()`.
- Looking at ConnectionHandlerImpl::GetDataOnSessionKey(), connection_key is broken into conn_handle and session_id by `PairFromKey()` then reconstructed into app_id with `KeyFromPair()`.

Also, if I look at perform_audio_pass_thru_request.cc, there are several places where `connection_key()` is used for app_id.

Actually, I have been confused why there are multiple variables "app_id", "connection_key" and "session_key" that eventually mean same thing. Perhaps, at the early stage of the project, they meant for different things? Anyway, my suggestion is to unify the args of `BeginAudioPassThru()` and `EndAudioPassThru()` to "app_id". I think this will clarify that only one app is running Audio Pass Thru at a time. Then, perhaps in audio_pass_thru_request.cc, we put a code something like
```
  uint32_t app_id = connection_key();
  application_manager_.BeginAudioPassThru(app_id);
```
which will hopefully avoid confusion.

Please let me know if it's OK for you.

Reply:
Thanks @jacobkeeler . I have renamed the args, and also updated the type to `uint32_t` to align with other codes. Hope this is OK.